### ISSUE TITLE
Sanitize the test file for verifying constraints

### ISSUE THREAD

**Issue by sdasgup3** [openxla/stablehlo#993]
This issue is specifically about re-factoring file [ops_stablehlo.mlir](https://github.com/openxla/stablehlo/blob/main/stablehlo/tests/ops_stablehlo.mlir) and thereby removing the associated technical debt. 

Improvements opportunities:
1. Alphabetically order the split tests based on the op name.   
2. Adopt a uniform format while naming the function, using the op name and intent of the tests (if applicable). Currently we have many functions with name `main` which does not convey what op is being tested or intent of the test. Suggestion: <op_name>_<XXX>. 
     - For negative tests, XXX = "a brief of the property being tested" OR `C*`, the constraint labels, from the spec
     - For Single positive test, XXX=""
     - For multiple positive tests, XXX=<counter> or anything else to distinguish the positive tests (there must be a reason for having multiple positive tests).
 3. Make sure all the split tests are have an associated "CHECK_LABEL". This can minimize the chances of get confusing log reports when something breaks.

Feel free to add more.

For (1)  
Minimum Prerequisite: Make sure that the all the function names starts with <op_name>. Names does not have to be as uniform as mentioned in (2). This will enable  (A) keeping all the tests w.r.t the same op in the same group, and  (B) make the groups sorted.

Once (1) is done, (2) can be seen as intra-group improvements.

**Comment by sdasgup3** [comment]
superseded by https://github.com/openxla/stablehlo/issues/1004
